scripts/plotting_tools.py

```diff
@@ -0,0 +1,329 @@
+    plt.colorbar(pcm_epi, ax=axs[1], fraction=0.042, pad=0.025)
+    
+    plt.tight_layout()
+    return axs
+
+
+def plot_winds(ds, base_plot, ptype=None):
+    '''plots u and v wind quivers on top of another plot'''
+    ax = base_plot(ds, ptype)
+    u = ds['u10'].values
+    v = ds['v10'].values
+    lat = ds['latitude'].values
+    lon = ds['longitude'].values
+    if isinstance(ax, np.ndarray): # case of multiple subplots (possibly add loop later)
+        ax[0].quiver(lon[::20, ::20], lat[::20, ::20], u[::20, ::20], v[::20, ::20], width=0.0015, transform=ccrs.PlateCarree())
+        ax[1].quiver(lon[::20, ::20], lat[::20, ::20], u[::20, ::20], v[::20, ::20], width=0.0015, transform=ccrs.PlateCarree())
+    else:
+        ax.quiver(lon[::20, ::20], lat[::20, ::20], u[::20, ::20], v[::20, ::20], width=0.0015, transform=ccrs.PlateCarree())
+    return ax
+    
+
+def plot_temp(ds, base_plot, ptype=None):
+    '''plots temp contours on top of another plot'''
+    ax = base_plot(ds, ptype)
+    lat = ds['latitude']
+    lon = ds['longitude']
+    temp_data = ds['t2m'] - 273.15  # Convert from Kelvin to Celsius
+    if isinstance(ax, np.ndarray):
+        for a in ax:
+            contour(a, lon, lat, temp_data)
+    else:
+        contour(ax, lon, lat, temp_data)
+    return ax
+    
+
+def plot_dpt(ds, base_plot, ptype=None):
+    '''plots dewpoint contours on top of another plot'''
+    ax = base_plot(ds, ptype)
+    lat = ds['latitude']
+    lon = ds['longitude']
+    dpt_data = ds['d2m'] - 273.15  # Convert from Kelvin to Celsius
+    
+    if isinstance(ax, np.ndarray):
+        for a in ax:
+            contour(a, lon, lat, dpt_data)
+    else:
+        contour(ax, lon, lat, dpt_data)
+    return ax
+
+
+def contour(ax, lon, lat, data):
+    contours = ax.contour(lon, lat, data, transform=ccrs.PlateCarree(), levels=10)
+    #zero_cel = ax.contour(lon, lat, dpt_data, transform=ccrs.PlateCarree(), levels=[0], colors='black', linewidths=2)
+    ax.clabel(contours, fontsize=10, colors='black')
+    #cbar = plt.colorbar(contours, orientation="horizontal", fraction=0.15, pad=0.025)
+    #cbar.set_label('Dew Pt Temperature (°C)')
+
+
+def plot_sp(ds, base_plot, ptype=None):
+    '''plot surface pressure contours on top of base plot'''
+    ax = base_plot(ds, ptype)
+    lat = ds['latitude']
+    lon = ds['longitude']
+    sp_data = ds['sp']
+
+    if isinstance(ax, np.ndarray):
+        for a in ax:
+            contour(a, lon, lat, sp_data)
+    else:
+        contour(ax, lon, lat, sp_data)
+    return ax
+
+
+def save(title, outname, ax=None):
+    print(f'saving {outname}')
+    if isinstance(ax, np.ndarray):
+        ax[0].set_title(title[0])
+        ax[1].set_title(title[1])
+    else:
+        plt.title(title)
+    plt.savefig(outname, bbox_inches='tight', dpi=300)
+    plt.close()
+
+
+if __name__ == '__main__':
+    logger = logging.getLogger(__name__)
+    logging.basicConfig(level=logging.INFO)
+
+    # Example usage:
```

Comment:
I would rename this script something along the lines of "create_plots.py" or something similar and only keep the main script here and import all of your other functions from a module in `./ptype/plotting.py` (or something similar) that way others can import the specific functions directly to be used for other analysis.